src/main/java/sirius/db/mongo/MongoQuery.java

```
@@ -131,6 +136,43 @@ public boolean exists() {
         return finder.selectFields(MongoEntity.ID).singleIn(descriptor.getRelationName()).isPresent();
     }
 
+    /**
+     * Returns a list of all items in the result, in a random order.
+     * <p>
+     * Internally, this uses a <tt>$sample</tt> aggregation with size equal to {@link #limit}.
+     * Note that large results should be processed using {@link #iterate(Function)} or
+     * {@link #iterateAll(Consumer)} as they are more memory efficient.
+     *
+     * @return a list of items in the query or an empty list if the query did not match any items
+     */
+    public List<E> randomList() {
+        List<E> result = new ArrayList<>();
+
+        // Ensure a sane limit...
+        if (limit > MAX_LIST_SIZE) {
+            throw Exceptions.handle()
+                            .to(Mixing.LOG)
+                            .withSystemErrorMessage("When using 'randomList' as most %s items can be selected. "
+                                                    + "Use 'iterate' for larger results. Query: %s",
+                                                    MAX_LIST_SIZE,
+                                                    this)
+                            .handle();
+        }
+
+        // Install circuit breaker...
+        if (limit == 0) {
+            limit = MAX_LIST_SIZE + 1;
```

shouldn't we rather throw an error here?